### PR TITLE
[UWP] Implement CollectionView Scrolled event 

### LINKED ISSUE #8508: [Double bug] UWP CollectionView.Scrolled not raised & unhandled exception on exit
### Description
UWP CollectionView.Scrolled not raised & unhandled exception on exit

### Steps to Reproduce
```xaml
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:d="http://xamarin.com/schemas/2014/forms/design"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             mc:Ignorable="d"
             x:Class="UWPCollectionViewScrolled.MainPage">
    <CollectionView Scrolled="CollectionView_Scrolled" x:Name="collectionView">
        <CollectionView.ItemsLayout>
            <LinearItemsLayout Orientation="Vertical" ItemSpacing="0" />
        </CollectionView.ItemsLayout>
        <CollectionView.ItemTemplate>
            <DataTemplate>
                <Label Text="{Binding}" HorizontalOptions="FillAndExpand"/>
            </DataTemplate>
        </CollectionView.ItemTemplate>
    </CollectionView>
</ContentPage>
```
```cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Xamarin.Forms;

namespace UWPCollectionViewScrolled
{
    // Learn more about making custom code visible in the Xamarin.Forms previewer
    // by visiting https://aka.ms/xamarinforms-previewer
    [DesignTimeVisible(false)]
    public partial class MainPage : ContentPage
    {
        public MainPage()
        {
            InitializeComponent();
            collectionView.ItemsSource = Enumerable.Range(1, 100);
        }

        private void CollectionView_Scrolled(object sender, ItemsViewScrolledEventArgs e)
        {
            System.Diagnostics.Debug.WriteLine("CollectionView_Scrolled: {0}", e.CenterItemIndex);
            System.Diagnostics.Debugger.Break();
        }
    }
}
```
1. Debug application
2. Scroll up and down
3. Close application

### Expected Behavior
1. Scrolled event is raised and debugger stops at the event handler.
2. Application does not generate an unhandled exception on exit.

### Actual Behavior
1. Scrolled event is not raised.
2. Application generates an unhandled exception on exit.

Exception has no stack trace when scrolling is not performed and immediately is closed when collectionView is visible.
Exception has following stack trace when scrolling is performed before closing:
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Xamarin.Forms.Application.SavePropertiesAsync()
   at Xamarin.Forms.Platform.UWP.WindowsBasePage.OnApplicationSuspending(Object sender, SuspendingEventArgs e)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__7_0(Object state)
   at System.Threading.WinRTSynchronizationContextBase.Invoker.InvokeCore()
```

### Basic Information

- Version with issue: 4.3
- Last known good version: N/A
- IDE: VS16.3.9
- Platform Target Frameworks: 
  - UWP:  17134
- Nuget Packages: Default packages from template
- Affected Devices: My PC

### Screenshots

image

### Reproduction Link

[UWPCollectionViewScrolled.zip](https://github.com/xamarin/Xamarin.Forms/files/3847166/UWPCollectionViewScrolled.zip)

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Implement CollectionView Scrolled event.

### Issues Resolved ### 

- fixes #8508

### API Changes ###
 
 None

### Platforms Affected ### 

- UWP

### Behavioral/Visual Changes ###

None

#### Before
issue8508-before
#### After
issue8508-after

### Testing Procedure ###
Launch Core Gallery and navigate to Issue8508. Scroll in the CollectionView below. If an alert appears, the Scrolled method has been invoked and the test has passed.

### PR Checklist ###

- [x] Targets the correct branch
- [x] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Platform.UAP/CollectionView/ItemsViewRenderer.cs [hunk 1]
```diff
@@ -387,5 +422,61 @@ protected virtual void UpdateEmptyViewVisibility()
...
+
+				itemsViewScrolledEventArgs.FirstVisibleItemIndex = firstVisibleItemIndex;
+				itemsViewScrolledEventArgs.CenterItemIndex = (lastVisibleItemIndex + firstVisibleItemIndex) / 2;
+				itemsViewScrolledEventArgs.LastVisibleItemIndex = lastVisibleItemIndex;
+			}
+
+			Element.SendScrolled(itemsViewScrolledEventArgs);
+		}
+
+		bool IsListViewItemVisibile(FrameworkElement element, FrameworkElement container)
```
Comment (velocitysystems) [inline Xamarin.Forms.Platform.UAP/CollectionView/ItemsViewRenderer.cs]: Looks good, just a minor spelling mistake: `IsListViewItemVisibile` --> `IsListViewItemVisible`.

**Comment by velocitysystems** [comment]
Great work on this @jsuarezruiz!

**Comment by samhouts** [comment]
@jsuarezruiz Also, please retarget to 4.4.0

File: Xamarin.Forms.Platform.UAP/CollectionView/ItemsViewRenderer.cs (hunk 1 above)
Comment (jsuarezruiz) [reply Xamarin.Forms.Platform.UAP/CollectionView/ItemsViewRenderer.cs]: Changed!. Thanks.

**Pull request merged**